src/coreComponents/unitTests/meshTests/testCornerPointMeshImport.cpp

```diff
@@ -0,0 +1,265 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+// Source includes
+#include "dataRepository/xmlWrapper.hpp"
+#include "mainInterface/GeosxState.hpp"
+#include "mainInterface/initialization.hpp"
+#include "codingUtilities/StringUtilities.hpp"
+#include "mesh/MeshManager.hpp"
+#include "mesh/generators/CellBlockManager.hpp"
+
+// special CMake-generated include
+#include "tests/meshDirName.hpp"
+
+// TPL includes
+#include <gtest/gtest.h>
+#include <conduit.hpp>
+
+using namespace geosx;
+using namespace geosx::dataRepository;
+
+void testMeshImport( string const & inputStringMesh,
+                     string const & inputStringRegion,
+                     string const & meshBodyName,
+                     string const & propertyToTest )
+{
+  conduit::Node node;
+  Group root( "root", node );
+  MeshManager meshManager( "mesh", &root );
+
+  // Load the mesh
+  xmlWrapper::xmlDocument xmlDocument;
+  xmlDocument.load_buffer( inputStringMesh.c_str(), inputStringMesh.size() );
+
+  xmlWrapper::xmlNode xmlMeshNode = xmlDocument.child( "Mesh" );
+  meshManager.processInputFileRecursive( xmlMeshNode );
+  meshManager.postProcessInputRecursive();
+  // we stop here for nonconfornming open vtk files
+  // file-file comparison
+  // write a function to parse and compare the generated files
+
+  // Create the domain and generate the Mesh
+  DomainPartition domain( "domain", &root );
+  meshManager.generateMeshes( domain );
+
+  // nonconforming stops here
+
+  MeshBody & meshBody = domain.getMeshBody( meshBodyName );
+  MeshLevel & meshLevel = meshBody.getMeshLevel( 0 );
+  NodeManager & nodeManager = meshLevel.getNodeManager();
+  FaceManager const & faceManager = meshLevel.getFaceManager();
+  ElementRegionManager & elemManager = meshLevel.getElemManager();
+
+  // Create the ElementRegions
+  xmlDocument.load_buffer( inputStringRegion.c_str(), inputStringRegion.size() );
+
+  xmlWrapper::xmlNode xmlRegionNode = xmlDocument.child( "ElementRegions" );
+  elemManager.processInputFileRecursive( xmlRegionNode );
+  elemManager.postProcessInputRecursive();
+
+  CellBlockManager & cellBlockManager = meshBody.getGroup< CellBlockManager >( keys::cellManager );
```

Comment:
I do not remember exactly, but could you do most of your test with a `CellBlockManagerABC` because this is eventually what you want to validate at the module interface.